exo-run/src/service-restarter.ls

```diff
@@ -10,37 +11,40 @@ require! {
 class ServiceRestarter extends EventEmitter
 
   ({@role, @service-location, @env, @logger}) ->
+    @stable-delay = 500
+    @debounce-delay = 500
     @docker-config-location = path.join process.cwd!, 'tmp'
+    @debounced-restart = debounce(@_restart, @debounce-delay)
 
 
   watch: ~>
     /* Ignores any sub-path including dotfiles.
     '[\/\\]' accounts for both windows and unix systems, the '\.' matches a single '.', and the final '.' matches any character. */
-    @watcher = watch @service-location, ignore-initial: yes, ignored: [/.*\/node_modules\/.*/, /(^|[\/\\])\../]
+    @watcher = watch @service-location, awaitWriteFinish: {stabilityThreshold: @stable-delay}, ignore-initial: yes, ignored: [/.*\/node_modules\/.*/, /(^|[\/\\])\../]
       ..on 'add', (added-path) ~>
         @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{added-path} was created"}
-        @_restart!
+        @debounced-restart!
       ..on 'change', (changed-path) ~>
         @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{changed-path} was changed"}
-        @_restart!
+        @debounced-restart!
       ..on 'unlink', (removed-path) ~>
         @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{removed-path} was deleted"}
-        @_restart!
+        @debounced-restart!
 
 
   _restart: ->
     @watcher.close!
-    cwd = @docker-config-location
+    cwd = @docker-config-location 
```

Comment:
Please remove the trailing whitespace. Ideally your editor should trim trailing whitespace when you save files.